src/uuid.c

```diff
@@ -70,6 +87,7 @@ int UUID_from_string(const char* uuid_string, UUID_T* uuid)
                     }
                     else
                     {
+                        uuid_bytes[j] = (char)(higherOrderDigit * 16 + lowerOrderDigit);
```

Comment:
nit: can this be bit shifted instead of multiplied? (since it looks like that's the goal)

Reply:
That's a good idea. Most compilers will be smart enough to optimize this though. But let's do it.

Reply:
Done.